tests/test_fingerprint.py

```diff
@@ -206,6 +208,15 @@ def func():
         self.assertEqual(hash1, hash3)
         self.assertNotEqual(hash1, hash2)
 
+        co_filename, returned_obj = os.path.sep.join([gettempdir(), "ipykernel_12345", "321456789.py"]), [0]
+        hash4 = md5(datasets.utils.py_utils.dumps(create_ipython_func(co_filename, returned_obj))).hexdigest()
+        co_filename, returned_obj = os.path.sep.join([gettempdir(), "ipykernel_12345", "321456789.py"]), [1]
+        hash5 = md5(datasets.utils.py_utils.dumps(create_ipython_func(co_filename, returned_obj))).hexdigest()
+        co_filename, returned_obj = os.path.sep.join([gettempdir(), "ipykernel_12345", "654123987.py"]), [0]
+        hash6 = md5(datasets.utils.py_utils.dumps(create_ipython_func(co_filename, returned_obj))).hexdigest()
+        self.assertEqual(hash4, hash6)
+        self.assertNotEqual(hash4, hash5)
```

Comment:
os.path.join also does the job ;)

```suggestion
        co_filename, returned_obj = os.path.join([gettempdir(), "ipykernel_12345", "321456789.py"]), [0]
        hash4 = md5(datasets.utils.py_utils.dumps(create_ipython_func(co_filename, returned_obj))).hexdigest()
        co_filename, returned_obj = os.path.join([gettempdir(), "ipykernel_12345", "321456789.py"]), [1]
        hash5 = md5(datasets.utils.py_utils.dumps(create_ipython_func(co_filename, returned_obj))).hexdigest()
        co_filename, returned_obj = os.path.join([gettempdir(), "ipykernel_12345", "654123987.py"]), [0]
        hash6 = md5(datasets.utils.py_utils.dumps(create_ipython_func(co_filename, returned_obj))).hexdigest()
        self.assertEqual(hash4, hash6)
        self.assertNotEqual(hash4, hash5)
```